src/output-json-alert.c

```diff
@@ -281,6 +299,19 @@ static int AlertJson(ThreadVars *tv, JsonAlertLogThread *aft, const Packet *p)
             }
         }
 
+        if (json_output_ctx->flags & LOG_JSON_DNS) {
+            if (p->flow != NULL) {
+                FLOWLOCK_RDLOCK(p->flow);
```

Comment:
This lock is no longer necessary now that https://github.com/inliniac/suricata/pull/2235 is merged.